### ISSUE TITLE
T10.15. Чи може клас мати кілька конструкторів?

### ISSUE THREAD

**Issue by WAG13** [kzhereb/knu-is-ooop2019#15]

**Comment by WAG13** [comment]
так

**Comment by i1red** [comment]
Так

**Comment by solovdaria** [comment]
Так, клас може мати безліч конструкторів, доки вони матимуть унікальні параметри.

**Comment by yanaskyrda** [comment]
Якщо ми визначаємо конструктор класу, який приймає параметри і хочемо залишити дефолтний конструктор, то потрібно про це прямо сказати компілятору:
`class A {`
`    std::string data;`
`public:`
`    A() = default;`
`    A (const std::string& newData) : data(newData) {}`
`}; `

**Comment by Velly69** [comment]
Так, а саме: 
- конструктор с параметрами
- конструктор за замовченням, який не приймає аргументи
- конструктор копіювання— конструктор, який приймає у якості аргумента об'єкт того ж класу (або посилання на нього)

**Comment by AntonHladkiy** [comment]
так , наприклад за замовчуванням та з параметрами

**Comment by RostyslavMV** [comment]
Може і часто має, часто прийнято створювати конструктори без параметрів, навіть якщо вони не викликаються. Крім конструктора без параметрів можна створити конструктори з різними наборами параметрів для заповнення конкретних полів в об'єкті, залежно від поставлених умов.

**Comment by ilyakorenevskyi** [comment]
Так це фактично є превантаження методу. Можна наприклад зробити конструктор за замовчуванням, що не приймає аргументів, або з аргументами
